src/hook/provider.tsx

```diff
@@ -1,25 +1,28 @@
 import React, { FC, useEffect, useRef, useState } from "react";
-import ToastContext from "./context";
+import ToastContext, { ToastType } from "./context";
 import Toast, { Props } from "../toast-container";
 
 type PropsWithChildren = Props & {
-    children: React.ReactNode;
+  children: React.ReactNode;
 }
 
+export let GlobalToast: ToastType
+
 const ToastProvider: FC<PropsWithChildren> = ({ children, ...props }) => {
-    const toastRef = useRef(null);
-    const [refState, setRefState] = useState({});
+  const toastRef = useRef(null);
```

Comment:
dont know if it helps, but you can type useRef and useState like useRef<string>()